Sources/HummingbirdWebSocket/WebSocketHandler.swift

```diff
@@ -130,62 +155,48 @@ actor WebSocketHandler: Sendable {
     }
 
     /// Send ping
-    func ping(outbound: NIOAsyncChannelOutboundWriter<WebSocketFrame>) async throws {
+    func ping(outbound: WebSocketOutboundWriter) async throws {
         guard !self.closed else { return }
         if self.pingData.readableBytes == 0 {
             // creating random payload
             let random = (0..<Self.pingDataSize).map { _ in UInt8.random(in: 0...255) }
             self.pingData.writeBytes(random)
         }
-        try await self.send(frame: .init(fin: true, opcode: .ping, data: self.pingData), outbound: outbound)
+        try await outbound.write(frame: .init(fin: true, opcode: .ping, data: self.pingData))
     }
 
     /// Send pong
-    func pong(data: ByteBuffer?, outbound: NIOAsyncChannelOutboundWriter<WebSocketFrame>) async throws {
+    func pong(data: ByteBuffer?, outbound: WebSocketOutboundWriter) async throws {
         guard !self.closed else { return }
-        try await self.send(frame: .init(fin: true, opcode: .pong, data: data ?? .init()), outbound: outbound)
+        try await outbound.write(frame: .init(fin: true, opcode: .pong, data: data ?? .init()))
     }
 
     /// Send close
     func close(
         code: WebSocketErrorCode = .normalClosure,
-        outbound: NIOAsyncChannelOutboundWriter<WebSocketFrame>,
+        outbound: WebSocketOutboundWriter,
         context: some WebSocketContextProtocol
     ) async throws {
         guard !self.closed else { return }
         self.closed = true
 
         var buffer = context.allocator.buffer(capacity: 2)
         buffer.write(webSocketErrorCode: code)
-        try await self.send(frame: .init(fin: true, opcode: .connectionClose, data: buffer), outbound: outbound)
-    }
-
-    /// Send WebSocket frame
-    func send(
-        frame: WebSocketFrame,
-        outbound: NIOAsyncChannelOutboundWriter<WebSocketFrame>
-    ) async throws {
-        var frame = frame
-        frame.maskKey = self.makeMaskKey()
-        try await outbound.write(frame)
-    }
-
-    /// Make mask key to be used in WebSocket frame
-    private func makeMaskKey() -> WebSocketMaskingKey? {
-        guard self.type == .client else { return nil }
-        let bytes: [UInt8] = (0...3).map { _ in UInt8.random(in: .min ... .max) }
-        return WebSocketMaskingKey(bytes)
+        try await outbound.write(frame: .init(fin: true, opcode: .connectionClose, data: buffer))
```

Comment:
I think we should consider making `close` non-throwing. I get that writing a frame can fail, but that's an _acceptable_ and ignorable failure in my book - at this point in the WebSocket's lifecycle. Possibly something to ask the NIO team's opinion on.

Reply:
I disagree here, this initiates the close handshake and doesn't actually close the websocket. It only gets closed when we receive the close connection message from the client. If we ignore the error then the close connection is never sent and we don't receive a close connection from the client. Remember also this is an internal function. Users will close the connection by exiting the handler.

In the current situation when the error is thrown out of the `asyncChannel.executeThenClose` closure which will force close the connection.

Reply:
Ah, good clarification. I misremember that then